proposals/extend_pod_security_admission.md

```diff
@@ -7,179 +7,226 @@
+          # +required
+          - controlName: Capabilities
+            images:
+              - ghcr.io/example/nginx:1.2.3
+            # +optional
+            restrictedField: spec.containers[*].securityContext.capabilities.add
+            # Any type
+            # +optional
+            values: # how to define an object?
+              - "KILL"
+              - "SETGID"
+              - "SETUID"
 ```
 
-Note that securityContext is configured at container level, not pod, as the matching image selects a specific container. The pod creation will be rejected if there's any violation found in its container. For pod level checks, other match criteria such as labels could be used.
+Note that securityContext is configured at container level, not pod, as the matching image selects a specific container. The pod creation will be rejected if there's any violation found in its container.
 
-For a PolicyReport, `cpol.rules.validate.podSecurity.exclude.restrictedField` and `cpol.rules.validate.podSecurity.exclude.controlName` will be added to `polr.results.properties` to audit the result.
+#### Controls at the pod spec level
+
+We use **labels** as a selector for pod security controls that are declared at the pod spec level.
+
+e.g.:
+
+- Host Namespaces: `spec.hostNetwork,spec.hostPID,spec.hostIPC`
+- Sysctls: `spec.securityContext.sysctls[*].name`
+- volumes: `spec.volumes`
+- Non-root groups: `spec.securityContext.supplementalGroups[*], spec.securityContext.fsGroup`
+
+In the following example we apply restricted PSS to namespaces "test" and "staging" and check the Control "Volumes" while excluding pods with the label "env: prod".
 
 ```yaml
-PolicyReport:
-results:
-- message: |- 
-    'validation error: The label `app.kubernetes.io/name` is required. 
-    Rule check-for-labels failed at path
-    /metadata/labels/app.kubernetes.io/name/'
-  policy: require-labels
-  resources:
-  - apiVersion: v1
-    kind: Pod
-    name: nginx
-    namespace: default
-    uid: 13fcb726-5597-4fcd-a708-ce8b558bd484
-  rule: check-for-labels
-  properties:
-    controlName: Capabilities
-    podSecurity: spec.containers[*].securityContext.capabilities.add
-  scored: true
-  status: fail
+validationFailureAction: enforce
+rules:
+  - name: enforce-restricted-exclude-all-volumes-prod
+    match:
+      any:
+        - resources:
+            kinds:
+              - Pod
+          namespaces:
+            - test
+            - staging
+    validate:
+      # this new type of rule only deals with PSS profiles
+      # as we need to check if the value in the resource
+      # is actually allowed by PSS
+      podSecurity:
+        profile: restricted
+        # version must be a valid Kubernetes minor version,
+        # or `latest`.
+        version: v1.24
+        exclude:
+          # controlName is the Control defined in PSS
+          # +required
+          - controlName: Volumes
+            labels:
+              env: prod
```

Comment:
The label should be under match/exclude block.

Reply:
fixed